File: 0763-partition-labels/0763-partition-labels.java
class Solution {
    public List<Integer> partitionLabels(String s) {
     HashMap<Character,Integer>h=new HashMap<>();
        for(int i=0;i<s.length();i++)
            h.put(s.charAt(i),i);
        int max=-1;
        int prev=-1;
        ArrayList<Integer>ans=new ArrayList<>();
        for(int i=0;i<s.length();i++)
        {
            max=(int)Math.max(max,h.get(s.charAt(i)));
            if(max==i)
            {
                ans.add(max-prev);
                prev=max;
            }
            
        }
        return ans;
    }
}

File: 0763-partition-labels/0763-partition-labels.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
     unordered_map<char,int>h;
        for(int i=0;i<s.length();i++)
            h[s[i]]=i;
        vector<int>arr;
        int maxi=-1,prev=-1;
        for(int i=0;i<s.length();i++)
        {
            maxi=max(maxi,h[s[i]]);
            if(maxi==i)
            {
                arr.push_back(maxi-prev);
                prev=maxi;
            }
        }
        return arr;
    }
};